models/Asset/WebDAV/Folder.php

```diff
@@ -50,39 +51,56 @@ public function getChildren()
     {
         $children = [];
 
-        if ($this->asset->hasChildren()) {
-            foreach ($this->asset->getChildren() as $child) {
-                if ($child->isAllowed('view')) {
-                    try {
-                        if ($child = $this->getChild($child)) {
-                            $children[] = $child;
-                        }
-                    } catch (\Exception $e) {
-                        Logger::warning($e);
-                    }
+        $db = Db::get();
```

Comment:
Use DI here:

```php
Pimcore\Db\ConnectionInterface $connection
```

Reply:
This is not a symfony service but it gets instantiated in https://github.com/pimcore/pimcore/blob/0c8b8e9ddc97dd5107a0a4f082857da2e651151c/bundles/AdminBundle/Controller/Admin/Asset/AssetController.php#L896

Of course we could change that but as there are a lot of other classes where `Db::get()` is used (like the above mentioned `AssetController`), I do not see a problem that this PR also does.

Reply:
I think it's fine for now.

Reply:
Never mind, now I provide the parameter separately for `setCondition()`, then we do not have to manually quote it.